src/Mvc/test/Mvc.FunctionalTests/AuthMiddlewareAndFilterTestBase.cs

```diff
@@ -0,0 +1,174 @@
+            await response.AssertStatusCodeAsync(HttpStatusCode.OK);
+        }
+
+        [Fact]
+        public async Task GlobalAuthFilter_CombinesWithAuthAttributeSpecifiedOnAction()
+        {
+            var action = "Authorized/ActionWithAuthAttribute";
+            var response = await Client.GetAsync(action);
+
+            await AssertAuthorizeResponse(response);
+
+            // LoginClaimA should be enough for the global auth filter, but not for the auth attribute on the action.
+            var authCookie = await GetAuthCookieAsync("LoginClaimA");
+
+            var request = new HttpRequestMessage(HttpMethod.Get, action);
+            request.Headers.Add("Cookie", authCookie);
+
+            response = await Client.SendAsync(request);
+            await AssertAuthorizeResponse(response);
+
+            authCookie = await GetAuthCookieAsync("LoginClaimAB");
+            request = new HttpRequestMessage(HttpMethod.Get, action);
+            request.Headers.Add("Cookie", authCookie);
+
+            response = await Client.SendAsync(request);
+            await response.AssertStatusCodeAsync(HttpStatusCode.OK);
+        }
+
+        [Fact]
+        public async Task AllowAnonymousOnPageConfiguredViaConventionWorks()
+        {
+            // Arrange & Act
+            var response = await Client.GetAsync("AllowAnonymousPageViaConvention");
+
+            // Assert
+            await response.AssertStatusCodeAsync(HttpStatusCode.OK);
+        }
+
+        [Fact]
+        public async Task AllowAnonymousOnPageConfiguredViaModelWorks()
+        {
+            // Arrange & Act
+            var response = await Client.GetAsync("AllowAnonymousPageViaModel");
+
+            // Assert
+            await response.AssertStatusCodeAsync(HttpStatusCode.OK);
+        }
+
+        [Fact]
+        public async Task GlobalAuthFilterAppliedToPageWorks()
+        {
+            // Arrange & Act
+            var response = await Client.GetAsync("PagesHome");
+
+            // Assert
+            await AssertAuthorizeResponse(response);
+
+            // We should be able to login with ClaimA alone
+            var authCookie = await GetAuthCookieAsync("LoginClaimA");
+
+            var request = new HttpRequestMessage(HttpMethod.Get, "PagesHome");
+            request.Headers.Add("Cookie", authCookie);
+
+            response = await Client.SendAsync(request);
+            await response.AssertStatusCodeAsync(HttpStatusCode.OK);
+        }
+
+        [Fact]
+        public Task GlobalAuthFilter_CombinesWithAuthAttributeOnPageModel()
+        {
+            // Arrange
+            var page = "AuthorizePageViaModel";
+
+            return LoginAB(page);
+        }
+
+        [Fact]
+        public Task GlobalAuthFilter_CombinesWithAuthAttributeSpecifiedViaConvention()
+        {
+            // Arrange
+            var page = "AuthorizePageViaConvention";
+
+            return LoginAB(page);
+        }
+
```

Comment:
Historically the place we see the most bugs is when there's a action that is trying to mix cookies and bearer, if its not too hard, do you think you can add one or two tests for that scenario here?

Basically something like:

```C#
[Authorize] // requires any authenticated user (aka the application cookie typically)
public class Controller {
   [Authorize(AuthenticationScheme = "Bearer")]
   public void Api();

   public void Cookie();

   [AllowAnonymous]
   public void PartyOn();
```

And ideally there would be a few tests that calls these actions at least twice in a few different orders (i.e. anon, bearer, cookie vs cookie, anon, bearer, etc) to make sure that we catch something like https://github.com/aspnet/AspNetCore/issues/7687